src/migrateWidgetsWithinDashboard.ts

```diff
@@ -0,0 +1,67 @@
+import { DataModel, Layout, getLayoutPath } from "@activeviam/activeui-sdk-5.1";
+import { _removeWidgetFromPage } from "@activeviam/dashboard-base-5.1";
+import _forEach from "lodash/forEach";
+import { MigrateWidgetCallback } from "./migration.types";
+import { WidgetFlaggedForRemovalError } from "./WidgetFlaggedForRemovalError";
+
+/**
+ * Mutates `dashboardState` by calling `callback` on each underlying widget state.
+ */
+export function migrateWidgetsWithinDashboard<
```

Comment:
I think it's good if we have a unit test for this function too.
I can handle it in a separate PR if you prefer.

Reply:
I added a quick one, but we could be more thorough in further PRs when focusing on testing.